File: restaurant/restaurant/page/restaurant/restaurant.py
import frappe,json


@frappe.whitelist()
def get_orders():
    orders = frappe.db.sql(""" SELECT * FROM `tabOrders` """, as_dict=1)

    for x in orders:
        x['table'] = '<table style="width: 100%;margin-top: 10px;border-bottom: 1px solid lightgray;margin-right: 10px" onclick="select_order({0})">'.format("'" + x.name + "'") + \
                     '<td style="width: 2%;border: 1px solid black;background-color: #00b0ff"></td>' \
                       '<td style="width: 40%;padding-left: 10px">#' + x.name.split("-")[1] + '<br>' + str(
            x.total_amount) + '<br>TABLE: ' + x.table + ' </td>' \
                                                        '<td style="width: 58%">' + str(x.posting_date) + " " + str(
            x.posting_time) + ' <br>' + x.mode_of_payment + ' <br>' + x.status + '</td>' \
                                                                                 '</table>'
        print(x['table'])
    selected_order = ""

    if len(orders) > 0:
        selected_order = get_selected_order(orders[0].name,[],from_get_orders=True)
    return {
        "orders": orders,
        "selected_order":selected_order,
        "dashboard": "change_tab('Dashboard')"
    }


@frappe.whitelist()
def get_selected_order(order_name,e_orders,from_get_orders=False):
    print(e_orders)
    orders = frappe.get_doc("Orders",order_name)
    items_table = ""
    items = frappe.db.sql(""" SELECT * FROM `tabOrder Items` WHERE parent=%s""", order_name, as_dict=1)
    for x in items:
        items_table += '<tr>'
        items_table += '<td style="padding: 15px">' + (x.item_name or "") + '</td>' \
                                                                            '<td style="padding: 15px">' + str(
            x.qty) + '</td>' \
                     ' <td style="padding: 15px">' + (x.notes or "") + '</td>' \
                                                                       ' <td style="padding: 15px">' + (
                       x.status or "") + '</td>' \
                                         ' <td style="padding: 15px">' + str(x.rate) + '</td>'
        items_table += '</tr>'
    selected_order = '<table style="width:100%;">' \
                      '<tr>' \
                      '<td style="padding: 10px" class="text-left">#' + orders.name.split("-")[1] + '</td>' \
                       '<td style="padding: 10px" class="text-right">KIOSK : Kiosk acticve</td>' \
                       '</tr>' \
                       '<tr>' \
                       '<td style="padding: 10px" class="text-left">' \
                       'Date and Time:<br>' + str(orders.posting_date) + " " + str(orders.posting_time) + \
                      '</td>' \
                      '<td style="padding: 10px" class="text-right">' \
                      'Payment Type:<br>' \
                      + orders.mode_of_payment + \
                      '</td>' \
                      '</tr>' \
                      '</table>' \
                      '' \
                      '<table style="width: 98%;margin: 1%">' \
                      ' <tr>' \
                      '<td style="padding: 10px" class="text-left">' \
                      ' Items' \
                      '</td>' \
                      '</tr>' \
                      '<th bgcolor="#d3d3d3" class="text-left" style="padding: 15px;width:20%">Item</th>' \
                      '<th bgcolor="#d3d3d3" class="text-left" style="padding: 15px;width:20%">Quantity</th>' \
                      ' <th bgcolor="#d3d3d3" class="text-left" style="padding: 15px;width:20%">Notes</th>' \
                      ' <th bgcolor="#d3d3d3" class="text-left" style="padding: 15px;width:20%">Status</th>' \
                      ' <th bgcolor="#d3d3d3" class="text-left" style="padding: 15px;width:20%">Price</th>' \
                      '</tr>' + items_table + \
                      '<th bgcolor="#d3d3d3" class="text-left" style="padding: 15px" colspan="4">Total</th>' \
                      '<th bgcolor="#d3d3d3" class="text-left" style="padding: 15px">' + str(orders.total_amount) + '</th>' \
                                  ' </tr>' \
                                  '</table>'

    return selected_order if from_get_orders else {
        "orders": json.loads(e_orders),
        "selected_order":selected_order,
        "dashboard": "change_tab('Dashboard')"
    }

@frappe.whitelist()
def create_order(values):
    data = json.loads(values)
    data['doctype'] = 'Orders'
    data['posting_date'] = frappe.utils.now_datetime().date()
    data['posting_time'] = frappe.utils.now_datetime().time()
    for x in data['order_item']:
        x['item_name'] = frappe.get_doc("Item",x['item']).item_name
        x['status'] = "Preparing"
    frappe.get_doc(data).insert()
    frappe.db.sql(""" UPDATE `tabTable` SET status='Occupied' WHERE name=%s """,data['table'])
    frappe.db.commit()
    return get_orders()

File: restaurant/restaurant/page/restaurant/restaurant.js
var page = ""
var orders = []
frappe.pages['restaurant'].on_page_load = function (wrapper) {
    page = frappe.ui.make_app_page({
        parent: wrapper,
        title: 'Restaurant',
        single_column: false
    });
    var advance_type = page.add_field({
        label: 'Order From',
        fieldtype: 'Select',
        fieldname: 'advance_type',
        options: ['', 'All', 'Table', 'Kiosk'],
        change() {

        }
    })
    var advance_type = page.add_field({
        label: 'Type',
        fieldtype: 'Select',
        fieldname: 'type',
        options: ['', 'Pay Waiter', 'Pay Counter', 'Pay POS', 'KNET'],
        change() {

        }
    })
    var advance_type = page.add_field({
        label: 'Order Number',
        fieldtype: 'Data',
        fieldname: 'order_number',
        change() {

        }
    })
    let $btn = page.set_primary_action('Create Order', () => create_order()
    )
    page.set_secondary_action('Refresh', () => fetch_data())

    fetch_data()
}

function fetch_data() {
    frappe.call({
        method: "restaurant.restaurant.page.restaurant.restaurant.get_orders",
        args: {},
        freeze: true,
        freeze_message: "Fetching Orders...",
        async: false,
        callback: function (resp) {
            orders = resp.message.orders
            resp.message['url'] =  "https://adminstaging.cocodine.me"
          if(page.sidebar[0].children.length > 0){
				cur_page.page.page.sidebar[0].replaceChild(document.createTextNode(""),cur_page.page.page.sidebar[0].children[0])
                page.sidebar.append(frappe.render_template("test", {'doc': resp.message}))
			} else {
                page.sidebar.append(frappe.render_template("test", {'doc': resp.message}))

			}

            if(page.main[0].children.length > 1){
				cur_page.page.page.main[0].replaceChild(document.createTextNode(""),cur_page.page.page.main[0].children[cur_page.page.page.main[0].children.length - 1])
                page.main.append(frappe.render_template("main", {'doc': resp.message}))
			} else {
                page.main.append(frappe.render_template("main", {'doc': resp.message}))

			}

        }
    });
    document.getElementById("navbar-search").style.display = "none"
    document.getElementsByClassName("navbar-nav")[1].style.display = "none"
    document.getElementsByClassName("search-icon")[0].style.display = "none"
    document.getElementsByClassName("page-head")[0].style.backgroundColor = "#C5CE92"
    document.getElementById("page-restaurant").style.backgroundColor = "#C5CE92"
}

function select_order(order) {
    frappe.call({
        method: "restaurant.restaurant.page.restaurant.restaurant.get_selected_order",
        args: {
            order_name: order,
            e_orders: orders
        },
        freeze: true,
        freeze_message: "Fetching Orders...",
        async: false,
        callback: function (resp) {

            orders = resp.message.orders
            resp.message['url'] =  "https://adminstaging.cocodine.me"
            if (page.main[0].children.length > 1) {

                console.log(page.main[0].children)
                cur_page.page.page.main[0].replaceChild(document.createTextNode(""),cur_page.page.page.main[0].children[cur_page.page.page.main[0].children.length - 1])
                page.sidebar[0].replaceChild(document.createTextNode(""), page.sidebar[0].children[0])

                page.sidebar.append(frappe.render_template("test", {'doc': resp.message}))
                page.main.append(frappe.render_template("main", {'doc': resp.message}))

            } else {

                page.sidebar.append(frappe.render_template("test", {'doc': resp.message}))
                cur_page.page.page.main[0].replaceChild(document.createTextNode(""),cur_page.page.page.main[0].children[cur_page.page.page.main[0].children.length - 1])

            }


        }
    });
}

function create_order() {
    var table_fields = [
        {
            fieldname: "item", fieldtype: "Link", options: "Item",
            in_list_view: 1, label: "Item"
        },
        {
            fieldname: "qty", fieldtype: "Float",
            in_list_view: 1, label: "Qty"
        },
        {
            fieldname: "rate", fieldtype: "Currency",
            in_list_view: 1, label: "Price",
        },
        {
            fieldname: "notes", fieldtype: "Small Text",
            in_list_view: 1, label: "Notes"
        }
    ];
    let d = new frappe.ui.Dialog({
        title: 'Enter details',
        fields: [
            {
                fieldname: "table", fieldtype: "Link", options: "Table",reqd:1,
                in_list_view: 1, label: "Table",
                get_query: function () {
                    return {
                        filters: {
                            status: "Available",
                            disabled: 0
                        }
                    }
                }
            },
            {
                fieldname: "mode_of_payment", fieldtype: "Link", options: "Mode of Payment",
                in_list_view: 1, label: "Payment Type"
            },

            {
                fieldname: "order_item",
                fieldtype: "Table",
                label: "Items",
                read_only: 1,
                cannot_add_rows: false,
                in_place_edit: false,
                data: [],
                fields: table_fields
            }
        ],
        size: 'small', // small, large, extra-large
        primary_action_label: 'Submit',
        primary_action(values) {
            frappe.call({
                method: "restaurant.restaurant.page.restaurant.restaurant.create_order",
                args: {
                    values:values
                },
                freeze: true,
                freeze_message: "Fetching Orders...",
                async: false,
                callback: function (resp) {
                    orders = resp.message.orders
                    resp.message['url'] =  "https://adminstaging.cocodine.me"
                    fetch_data()
                    d.hide()

                }
            });
        }
    });
    d.show();
    console.log("HERE")
    setTimeout(()=>{
        document.getElementsByClassName("modal-content")[document.getElementsByClassName("modal-content").length-1].style.width = "700px"
    },150)

}

function change_tab(tab) {
    console.log("============")
    console.log(tab)
}

function void_order(name) {
    frappe.confirm('Are you sure you want to proceed?',
    () => {
        frappe.call({
            method: "restaurant.restaurant.page.restaurant.restaurant.void_order",
            args: {
                order_name: name,
            },
            freeze: true,
        freeze_message:"Voiding order.....",
        callback: function () {
             fetch_data()
        }
        })
    }, () => {
        // action to perform if No is selected
    })
}
function paid_order(name) {
    frappe.confirm('Are you sure you want to proceed?',
    () => {
        frappe.call({
            method: "restaurant.restaurant.page.restaurant.restaurant.paid_order",
            args: {
                order_name: name,
            },
            freeze: true,
        freeze_message:"Paid order.....",
        callback: function () {
             fetch_data()
        }
        })
    }, () => {
        // action to perform if No is selected
    })
}